java/rocksjni/portal.h

```diff
@@ -6779,14 +6779,14 @@ class ColumnFamilyDescriptorJni : public JavaClass {
     }
 
     jmethodID mid = env->GetMethodID(jclazz, "<init>",
-                                     "([BLorg/rocksdb/ColumnFamilyOptions;)V");
+                                     "([BLorg/rocksdb/ColumnFamilyOptions;Z)V");
```

Comment:
Do we still create Java ColumnFamilyDescriptor objects from C++, I would think not after this refactoring...

Reply:
When we ask for a CFD for the CFH, there is no Java CFD in the CFH,
so we are pretty much obliged to create a new one.

Which is really me agreeing that we need to do (2).

Anyway,
the issue https://github.com/facebook/rocksdb/issues/12224
is about getting the descriptor for the column family handle, 
and the safest thing to do is to create an up-to-date descriptor 
as we do now. On the C++ side a new CFO is copied from the CFO
which is held in the CFD when getOptions, so that is "safe" for
deletion from Java when the owning CFD is deleted.

I wonder if it would be possible to never have the Java CFD hold any
Java CFO ? Just use a default one on construction if a non-default one
is not supplied, and create a new one overJNI/C++ when getOptions()
is called ? Then you would get rid of the `implicitlyCreated` complexity.
What do you think ?

Reply:
We also need to consider two things:

1. Ownership
2. Expected behaviour - if we return a new object each time - is it equal to the previous object and does the user know that they have to dispose each new instance?

Reply:
As I understand it, by making the ColumnFamilyDescriptor a RocksObject we are giving it to the caller, and it is now their responsibility to close it. Previously they should have needed to "close" it but they couldn't do that.
I think my subsidiary proposal is arguably more consistent in that the returned options are always owned by the caller and it is a simple rule that they need to close them. But if we don't do that they would be `implicitlyCreated`
and so closing the ColumnFamilyDescriptor would be enough. So maybe I'm arguing that the code as Radek has it now is about as good as we can get. Should we have

1. A statement in the changelog that `ColumnFamilyDescriptor`s are now autocloable and should be closed ?
2. A test that the usage pattern in the ticket wrapped in a try-with-resources doesn't leak ?